### PR TITLE
Add Primitives to DIVE.

### PR DESCRIPTION
**Pull request by feliopterix**
### 5. Checklist

- [ ] I have rebased my changes to remove merge conflicts
- [ ] I have written tests and verified that they fail without my change
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [ ] I have read the contribution requirements and fulfil them.

### CODE REVIEW

**Review by dfrancos-hub — DISMISSED** [review]

File: src/primitive/Primitive.ts [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+            3, 2, 4,
+            3, 0, 4,
+            0, 1, 2,
+            0, 2, 3,
+        ], 1));
+        return geo;
+    }
+
+    private createBoxGeometry(geometry: COMGeometry): BufferGeometry {
+        const geo = new BoxGeometry(geometry.width, geometry.height, geometry.depth);
```
Comment (dfrancos-hub) [inline src/primitive/Primitive.ts]: Approved, just a side note I guess some of the lasts createXZYGeometry functions can return directly, bypassing the constant creation as the output is clear enough 👍 .

File: src/com/types.ts [hunk 2]
```diff
@@ -28,4 +28,27 @@ export type COMModel = COMBaseEntity & {
     loaded: boolean;
 };
 
-export type COMEntity = COMPov | COMLight | COMModel;
+export type COMGeometry = {
+    name: string
+    width: number;
+    height: number;
+    depth: number;
```
Comment (Deristes) [inline src/com/types.ts:35]: @ffrank913
Just a small question, why do we need to define a width, height and depth for a Primitive Geometry.
Isn't that the same as just using 1, 1, 1 and scaling it via the scale vector, or am i getting this wrong?

Comment (feliopterix) [reply src/com/types.ts:35]: The definition for the geometry gives you information about the initial size of a given shape (defined by `name`and interpreted by DIVE in a later step). We need to have some kind of definition for the base geometry.

Counter-question (what is Gegenfrage in english? kekw): How would you tell me what the radius of a cylinder is by only having a scale value?

File: src/primitive/Primitive.ts (hunk 1 above)
Comment (feliopterix) [reply src/primitive/Primitive.ts]: Yep, true, I will change this! :)

Comment (feliopterix) [reply src/primitive/Primitive.ts]: Please check again, thank you in advance!

Comment (feliopterix) [reply src/primitive/Primitive.ts]: btw I just changed that given approvals are revoked then pushing something after the approval

**Review by dfrancos-hub — APPROVED** [review]

**Review by Deristes — APPROVED** [review]

**Pull request merged**